src/clr-angular/data/datagrid/datagrid-items.ts

```diff
@@ -12,24 +12,38 @@ import {
     OnChanges,
     SimpleChanges,
     TemplateRef,
-    TrackByFunction
+    TrackByFunction,
+    ViewContainerRef
 } from "@angular/core";
 
 import {Items} from "./providers/items";
 
+export type RowContext<T> = {
+    $implicit: T
+};
+
 @Directive({
     selector: "[clrDgItems][clrDgItemsOf]",
 })
-export class ClrDatagridItems implements OnChanges, DoCheck {
-    private _rawItems: any[];
+export class ClrDatagridItems<T> implements OnChanges, DoCheck {
+    private _rawItems: T[];
     @Input("clrDgItemsOf")
-    public set rawItems(items: any[]) {
+    public set rawItems(items: T[]) {
         this._rawItems = items ? items : [];
     }
     private _differ: IterableDiffer<any>;
-
-    constructor(public template: TemplateRef<any>, private _differs: IterableDiffers, private _items: Items) {
+    constructor(public template: TemplateRef<RowContext<T>>, private _differs: IterableDiffers, private _items: Items,
+                private vcr: ViewContainerRef) {
         _items.smartenUp();
+        _items.change.subscribe(items => {
+            const displayed = items;
+            this.vcr.clear();  // It is additive if I don't clear the datagrids vcr
+            // Will need to use IterableDiff to properly account for trackBy
```

Comment:
Yeah, this is going to be a performance issue. My best recommendation is to check out the code for `NgForOf` itself, and I'm okay with you copying most of it. 

Or even better, maybe you can see if you can simply proxy to an `NgForOf` that you instantiate manually, that would be ideal. Ping me if you want to discuss it, I'd be happy to pair for a little bit to help you get that done.

Reply:
Thanks for the `NgForOf` idea. Will address this with that.